unlock-app/src/components/content/certification/CertificationDetails.tsx

```diff
@@ -1,3 +1,269 @@
-export const CertificationDetails = () => {
-  return null
+import { useQuery } from '@tanstack/react-query'
+import Link from 'next/link'
+import { ReactMarkdown } from 'react-markdown/lib/react-markdown'
+import { useMetadata } from '~/hooks/metadata'
+import { toFormData } from '~/components/interface/locks/metadata/utils'
+import { Button, Detail, Icon, Placeholder } from '@unlock-protocol/ui'
+import router from 'next/router'
+import { useLockManager } from '~/hooks/useLockManager'
+import { RiExternalLinkLine as ExternalLinkIcon } from 'react-icons/ri'
+import { networks } from '@unlock-protocol/networks'
+import { addressMinify } from '~/utils/strings'
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+import { expirationAsDate } from '~/utils/durations'
+import { IoLogoLinkedin as LinkedinIcon } from 'react-icons/io'
+import { RiDownloadLine as DownloadIcon } from 'react-icons/ri'
+
+interface CertificationDetailsProps {
+  lockAddress: string
+  network: number
+  tokenId: string
+}
+
+export const CertificationDetails = ({
+  lockAddress,
+  network,
+  tokenId,
+}: CertificationDetailsProps) => {
+  const { data: metadata, isInitialLoading: isMetadataLoading } = useMetadata({
+    lockAddress,
+    network,
+  })
+
+  const { isLoading, data: key } = useQuery(
```

Comment:
I suspect we're already doing that in many places. Please DRY up the code and just introduce a new hook for that.